custom/icds_reports/static/js/angular-services/baseControllers.service.js

```diff
@@ -15,6 +15,10 @@ window.angular.module('icdsApp').factory('baseControllersService', function() {
                 $location.search('location_name', loc.name);
             }
         };
+        vm.selectedLocationsCount = function() {
+              return (vm.selectedLocationLevel===4) ? (vm.selectedLocations.filter(Boolean).length) :
```

Comment:
what does `vm.selectedLocations.filter(Boolean)` do?

Reply:
selected Locations is an array of fixed size of 5. when there are few selected locations, remaining entries will be null. this removes null values and resize the array